---
title: "AuPairWise-vignette"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{AuPairWise-vignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Overview:
AuPairWise is a bioinformatics tool that evaluates the replicability of an RNA-seq experiment, based on the robustness of the expression data rather than the quality control measures and procedures used in the lab.

AuPairWise makes this evaluation using input from two sources: the raw expression data from the experiment, and a list of gene pairs which are known to be co-expressed. 

The tool performs a series of runs, testing how effective the RNA-seq data is at correctly classifying the co-expression of its genes, based on the known data. Each set of runs generates a ROC curve. The area under that curve, called the AUROC, quantifies the classification ability of the data. An AUROC of 0.5 is expected for a completely random classifier, while an AUROC of 0.75 or higher is generally indicative of a good classifier.

AuPairWise generates an AUROC for increasing levels of noise in the data. As the noise level increases, a good set of data will be more able to discriminate between normal and perturbed samples against the known co-expressed pair of genes, than against a set of random gene pairs.

The key output of AuPairWise is a summary showing the generated AUROCs for each noise factor. P-values are also calculated to assess the evaluation generated by AuPairWise These are given by the difference between the observed AUROCs for each noise factor, and AUROC for the random pairs, which act as a NULL.

## Setting Up
To use AuPairWise, you must have installed R and Bioconductor. Vist http://bioconductor.org for installation instructions.
To install the AuPairWise package, evaluate the following commands in an R session:
```
if (!require("BiocManager"))
  install.packages("BiocManager")
BiocManager::install("AuPairWise")
```

## Using AuPairWise
The first thing you must do is load AuPairWise using `libary` or `require`.
```
library(AuPairWise)
```

There are three required arguments to `runAPW()`:

1. A BioBase ExpressionSet object. Refer to https://www.bioconductor.org/packages/devel/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf for help understanding this object.
1. A path to a directory used for results
1. A list of pairs of genes that are known to be co-expressed. Use either a custom set, or the stoich.pairs set found in `AuPairWise/SampleData/pairs.Rdata`.

Below is a typical workflow that will get you started with AuPairWise.
```
library(AuPairWise)
library(Biobase)

# Set a working directory
masterdir = "~/Path/To/Working/Directory"

# Load expression data
load(paste(masterdir,"/SampleData/sample_brainspan.Rdata",sep="")) # TODO: we should load this from the /data directory in the package

# Load stoich pair data
load(paste(masterdir,"/SampleData/pairs.Rdata",sep="")) 

# Setting output directory
out = "~/Path/To/Directory/Where/You/Would/Like/Results/To/Be/Placed"

# Generate Biobase ExpressionSet object from expression data
minimalExpressionSet <- ExpressionSet(assayData=exprs)

# Run AuPairWise itself
summary = run_APW(minimalExpressionSet, out, stoich.pairs)
```

You may optionally specify the noise factors and number of repeats used in the evaluation.
By default, these are `n.factors=c(0,1,5,10,20,50,100)` and `n.repeats=10`. Below is an example of `run_APW()` being called with custom parameters, with fewer noise factors and fewer repeats than the default.
```
summary = run_APW(minimalExpressionSet, out, stoich.pairs, n.factors=c(0,5,20,50,100), n.repeats=3)
```

This code may be used to generate a snapshot of the data in a short time. However, a reliable evaluation should cover at minimum all default noise factors and use at least 10 repeats.





