---
title: "AuPairWise-vignette"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{AuPairWise-vignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(AuPairWise)
```

## Overview:
AuPairWise is a bioinformatics tool that evaluates the replicability of an RNA-seq experiment, based on the robustness of the expression data rather than the quality control measures and procedures used in the lab.

AuPairWise makes this evaluation using input from two sources: the raw expression data from the experiment, and a list of gene pairs which are known to be co-expressed. 

The tool performs a series of runs, testing how effective the RNA-seq data is at correctly classifying the co-expression of its genes, based on the known data. Each set of runs generates a ROC curve. The area under that curve, called the AUROC, quantifies the classification ability of the data. An AUROC of 0.5 is expected for a completely random classifier, while an AUROC of 0.75 or higher is generally indicative of a good classifier.

AuPairWise generates an AUROC for increasing levels of noise in the data. As the noise level increases, a good set of data will be more able to discriminate between normal and perturbed samples against the known co-expressed pair of genes, than against a set of random gene pairs.

The key output of AuPairWise is a summary showing the generated AUROCs for each noise factor. P-values are also calculated to assess the evaluation generated by AuPairWise These are given by the difference between the observed AUROCs for each noise factor, and AUROC for the random pairs, which act as a NULL.

