---
title: "AuPairWise-vignette"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{AuPairWise-vignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Overview
AuPairWise is a bioinformatics tool that evaluates the replicability of an RNA-seq experiment, based on the robustness of the expression data rather than the quality control measures and procedures used in the lab.

AuPairWise makes this evaluation using input from two sources: the raw expression data from the experiment, and a list of gene pairs which are known to be co-expressed. 

The tool performs a series of runs, testing how effective the RNA-seq data is at correctly classifying the co-expression of its genes, based on the known data. Each set of runs generates a ROC curve. The area under that curve, called the AUROC, quantifies the classification ability of the data. An AUROC of 0.5 is expected for a completely random classifier, while an AUROC of 0.75 or higher is generally indicative of a good classifier.

AuPairWise generates an AUROC for increasing levels of noise in the data. As the noise level increases, a good set of data will be more able to discriminate between normal and perturbed samples against the known co-expressed pair of genes, than against a set of random gene pairs.

The key output of AuPairWise is a summary showing the generated AUROCs for each noise factor. P-values are also calculated to assess the evaluation generated by AuPairWise These are given by the difference between the observed AUROCs for each noise factor, and AUROC for the random pairs, which act as a NULL.

## Paper
Ballouz S, Gillis J (2016) AuPairWise: A Method to Estimate RNA-Seq Replicability through Co-expression. PLoS Comput Biol 12(4): e1004868. https://doi.org/10.1371/journal.pcbi.1004868

## Setting Up
To use AuPairWise, you must have installed R and Bioconductor. Vist http://bioconductor.org for installation instructions.
To install the AuPairWise package, evaluate the following commands in an R session:
```
if (!require("BiocManager"))
  install.packages("BiocManager")
BiocManager::install("AuPairWise")
```

## Using AuPairWise
The first thing you must do is load AuPairWise using `libary` or `require`.
```
library(AuPairWise)
```

There are three required arguments to `runAPW()`:

1. A BioBase ExpressionSet object. Refer to https://www.bioconductor.org/packages/devel/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf for help understanding this object.
1. A path to a directory used for results
1. A list of pairs of genes that are known to be co-expressed, labelled by their entrez gene IDs. Use either a custom set, or the ```samplePairs``` set found in this package. 

Below is a typical workflow that will get you started with AuPairWise. This workflow uses two in-built sample datasets in the AuPairWise package:

* Brainspan RNA-seq data, as an ExpressionSet object
* Sample pair data

You may choose to use the provided pair data in your workflow in many cases, though you should replace the RNA-seq data with your own.

```
library(AuPairWise)

# Set a working directory
masterdir = "~/Path/To/Working/Directory"

# Loading expression data
data("sampleBrainspanExpressionSet")

# Loading pair data
data("samplePairs")

# Setting output directory
out = "~/Path/To/Directory/Where/You/Would/Like/Results/To/Be/Placed"

# Run AuPairWise itself
summary = run_APW(sampleBrainspanExpressionSet, out, samplePairs)
```

You may optionally specify the noise factors and number of repeats used in the evaluation.
By default, these are `n.factors=c(0,1,5,10,20,50,100)` and `n.repeats=10`. Below is an example of `run_APW()` being called with custom parameters, with fewer noise factors and fewer repeats than the default.
```
summary = run_APW(sampleBrainspanExpressionSet, out, samplePairs, n.factors=c(0,5,20,50,100), n.repeats=3)
```

This code may be used to generate a snapshot of the data in a short time. However, a reliable evaluation should cover at minimum all default noise factors and use at least 10 repeats.


## Interpreting the output

### Results:
The output of the ``` run_APW()``` function is a list that contains 6 elements. Here, we've named it summary.
The first element is a table with the average AUROCs for the noise factors that the model ran: ``` summary$aurocs ```
The first row of this table has the stoichiometric pair results, and the second row has the random pairs run.
The next two lists are the statistics for the above AUROCs (standard deviation and standard error): ``` summary$aurocs.sd ```  and ``` summary$aurocs.se```
The noise factors tested are listed in ```summary$n.factors```. These are the default values, and can 
be set in the ```run_APW()``` function, eg. ``` run_APW(sampleBrainspanExpressionSet, out, samplepairs, n.factors=c(0,10,50) )```
The p-valaue of the Wilcoxon rank sum test between the stoichiometric pairs and the random pairs for each noise factor are in ```summary$pvals```
And finally, the estimated noise factors for any AUROC is given in: ``` summary$stats```

|AUROCs|0|1|2|5|10|15|20|25|50|100|
|---|---|---|---|---|---|---|---|---|---|---|
|Stoichiometric pairs|0.59|0.59|0.54|0.58|0.82|0.90|0.99|0.99|1.00|1.00|
|Random|0.59|0.49|0.55|0.59|0.74|0.79|0.88|0.95|1.00|1.00|


|P-values|0|1|2|5|10|15|20|25|50|100|
|---|---|---|---|---|---|---|---|---|---|---|
|Ranksum test|0.9096|0.2242|1.0000|0.9698|0.1381|0.0598|0.0025|0.0246|NaN|NaN|

### Visualizing results:
You can view the summary results with the ``` plot_summary_results(summary)``` function.

![Sample output](https://github.com/sarbal/AuPairWise/blob/master/suppl/imgs/summary_encode.png "Sample output")
 
### Intepreting results
A low noise factor implies that your experiment 'replicates' the coexpression patterns expected well
enough to be detected by small perturbations.
However, this should also be compared to the performances given by the random pairs.

### Extras 
To run on a different list of gene pairs, modify the ```samplePairs``` variable.
This a matrix of gene pairs, the first column has gene A and second column gene B.
These need to be entrez gene IDs, or must match the gene labels of your expresssion dataset.

You can also combine multiple runs, but this requires some careful labelling of your output files. 



